tests/js/client/shell/shell-admin-cluster-rebalance.js

```diff
@@ -338,6 +335,75 @@ function clusterRebalanceOtherOptionsSuite() {
   };
 }
 
+function clusterRebalanceWithMovesToMakeSuite() {
+  let prevDB = null;
+  const cn = "col1";
+
+  return {
+
+    setUpAll: function() {
+      prevDB = db._name();
+      db._createDatabase(database);
+      db._useDatabase(database);
+    },
+
+    tearDownAll: function() {
+      db._useDatabase(prevDB);
+      db._dropDatabase(database);
+    },
+
+
+    testCalcRebalanceAfterUnbalanced: function() {
+      for (let i = 1; i <= 3; ++i) {
+        const toServer = "DBServer000" + i;
+        db._create(cn, {numberOfShards: 10, replicationFactor: 1});
+        const plan = arango.GET("/_admin/cluster/shardDistribution").results[cn].Plan;
+        Object.entries(plan).forEach((shardInfo) => {
+          const [shardName, servers] = shardInfo;
+          const leader = servers.leader;
+          if (leader === toServer) {
+            return;
+          }
+          let moveShardJob = {
+            database: database,
+            collection: cn,
+            shard: shardName,
+            fromServer: leader,
+            toServer: toServer,
+            isLeader: true,
+            remainsFollower: false
+          };
+          const result = arango.POST("/_admin/cluster/moveShard", moveShardJob);
+          assertEqual(result.code, 202);
+          let start = internal.time();
+          while (true) {
+            if (internal.time() - start > 120) {
```

Comment:
Worst case, this test takes 120 s * 10 shards * 3 servers = 3600 s = 1h.
I think we should probably use an overall time budget here, e.g. 300 seconds, for the *entire* test.